### PR TITLE
Schedule Gantt

### PR DESCRIPTION
**Pull request by ruebel**
This is showing test data from the server for now where everyone's work hours are 9-5 in their timezone.  @jdeal has a forthcoming PR that will add real data to that call.  I don't believe we will have to change anything on the frontend to accommodate that new data but I thought I would point it out.

@jdaudier I would love if you could take a pass at this and add your own styles to it.  I think the site has great styles so far and we were more focused on getting the data to render and didn't give the style as much QT as it needed.  I can always come back through and do another pass on the styles if you don't want to.  Please let me know if you have questions, this ended up being a pretty serious refactor in parts of the code.

✨ Adds Schedule Gantt view
♻️ Refactors filters to be generic
✨ Adds mock data for local development without server
✨ Adds `zapier-scripts`, `emotion`

### CODE REVIEW

**Comment by jdaudier** [comment]
Thanks @ruebel! I will likely review this over the Thanksgiving weekend.

This is me sometimes...

File: src/components/Accordion.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ export default class Accordion extends Component {
 
     const totalCheckedFilters = Object.values(filters).reduce(
       (acc, filter) => acc + filter.length,
-      {}
+      0
```
Comment (jdaudier) [inline src/components/Accordion.js]: The pink dot will pulsate if there are filters selected and the accordion is closed.

File: src/components/Home.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Home extends Component {
     isMapLoaded: false,
     isSidePanelOpen: false,
     isAboutOpen: false,
-    isGantOpen: false,
+    isGanttOpen: false,
```
Comment (jdaudier) [inline src/components/Home.js]: Fixed typo on the word `Gantt`.

File: src/css/Accordion.css [hunk 3]
```diff
@@ -5,7 +5,7 @@
     right: 20px;
     width: 220px;
     transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
-    z-index: 1001;
+    z-index: 1;
```
Comment (jdaudier) [inline src/css/Accordion.css]: Accordion has to be below the Header in order to fix this bug

## BEFORE


## AFTER

**Comment by jdaudier** [comment]
@ruebel Thanks for working on this enhancement!

**A couple of questions...**

1. Can you explain the reasoning for the time to start with **21:00, 22:00**, and so on? I figure that'll give me some context.

	
 
2. Is there a certain order for the names? It seems to be alphabetical by last name, but it seems to restart in different sections.

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": [["zapier"]]
```
Comment (jdaudier) [inline .babelrc]: Just wanted to confirm that we need a nested array here.

Comment (ruebel) [reply .babelrc]: I've always seen it like this, looks silly, but works

**Comment by jdeal** [comment]
@ruebel @jdaudier I added a few bits to this including:

- Split up into "fast" and "full" endpoints so that Mapier proper loads really fast, but the schedule view loads when the Slack data is ready.
- Retrying requests in case the mapier-api craps out. Which it will for longer request, but this also helps during restarts or whatever flakiness might occur.
- Show gray 9-5 bars if we don't have working hours for someone, so there's at least _something_. Since most people work 9-5ish, I figure that's better than nothing, and the gray tells you it's just a guess.
- Had to muck with the loading indicator a bit to make it work in the schedule context.
- Added a hail mary alert for if requests fail. It's not fun to have it just sit and spin forever if something goes wrong. Now (after retries), it will alert you to tell you something went wrong.

**Comment by ruebel** [comment]
@jdaudier @jdeal I think this is at the point where it meets the minimal viable product standard and we should merge it.  It's not perfect but I think it's to the point where our users could start getting value out of it.  I would say we merge this and then wait for the monorepo change to happen before we do anything else.  Who's with me?

**Comment by jdeal** [comment]
@ruebel I'm 👍 to merge.

File: src/components/Loader.js [hunk 5]
```diff
@@ -1,19 +1,25 @@
...
-                <div className="loader--dot" />
-                <div className="loader--dot" />
-                <div className="loader--text" />
-            </div>
+const Loader = ({ isDark, shouldBlockEvents = true, children }) => {
+  const loaderDarkModifier = isDark ? 'loader--dark' : '';
+  return (
+    <div
+      className="loader-container"
+      style={{ pointerEvents: shouldBlockEvents ? '' : 'none' }}
```
Comment (jdaudier) [inline src/components/Loader.js]: @jdeal I just converted the loader to use EmotionJS, and I reverted this line.

Please LMK if that's incorrect.

`pointerEvents: shouldBlockEvents ? 'none' : 'auto',`

**Pull request merged**